### DISCUSSION TITLE
How to use `pscale connect` for testing?

### DISCUSSION THREAD

**Discussion by janpio** [planetscale/discussion#33] [Q&A] [answered]
We use `pscale connect e2e-tests main --execute-protocol 'mysql' --execute 'yarn test'` to open a connection to the database, then run our application tests `yarn test` that uses the provided `DATABASE_URL`. That all works great, but at the end the `pscale` process keeps running indefinitely and we need to kill it somehow:

```
Local address to connect your application: 127.0.0.1:3306 (press ctrl-c to quit)
yarn run v1.22.10
warning ../../package.json: No license field
$ jest --forceExit
...
# runs forever now, waiting for ctrl-c
```

Is there a way for `pscale connect` to automatically stop/shut down after the executed command finishes?

**Comment by aquarapid** [comment]
A bit of a hack, but using a wrapper script that kills the parent (`pscale connect`) works:

```
$ cat /tmp/wrapper.sh 
#!/bin/bash
<do something that eventually exits here>
kill $PPID

$ chmod +x /tmp/wrapper.sh

$ pscale connect mydatabase mybranch --debug --execute '/tmp/wrapper.sh'
2021-05-22T15:10:07-07:00       info    adding tls.Config to the cache  {"instance": "myorg/mydatabase/mybranch"}
2021-05-22T15:10:07-07:00       info    ready for new connections
2021-05-22T15:10:07-07:00       info    listening remote DB instance    {"local_addr": "127.0.0.1:3306", "instance": "myorg/mydatabase/mybranch"}
Secure connection to databases mydatabase and branch mybranch is established!.

Local address to connect your application: 127.0.0.1:3306 (press ctrl-c to quit)2021-05-22T15:10:17-07:00       info  received context cancellation, waiting until timeout     {"timeout": "1s"}
2021-05-22T15:10:17-07:00       info    no connections to wait, bailing out
skipping update, reason: Latest version was already checked
$
```

We'll add a feature request for an option for connect to exit after the execute progam exits.

**Reply by janpio** [reply]
Thanks, that fundamentally works I think - but the tests are now failing as the script executing that command exits with status code 1 (`test.sh` => `pscale connect ... --execute './test-wrapper.sh'` => `yarn test`):

https://github.com/prisma/e2e-tests/blob/planetscale/databases/planetscale/test.sh
https://github.com/prisma/e2e-tests/blob/planetscale/databases/planetscale/test-wrapper.sh

Any additional shell script magic to make this report the actual `yarn test` result anyway?
Does command this kill the `pscale` command or my `test.sh` script btw?

**Comment by fatih** [comment, marked as answer]
I changed the behavior with this PR: https://github.com/planetscale/cli/pull/278 Now it'll exit and close the internal proxy (which means `pscale` will exit as well). If you have a chance to use it let me please know.

**Reply by janpio** [reply]
Thanks, that works fine so far - it indeed ends the process.

Follow up request: https://github.com/planetscale/cli/issues/283
